### ISSUE TITLE
Update documentation 

### ISSUE THREAD

**Issue by carstingaxion** [GatherPress/gatherpress#656]
### Is your enhancement related to a problem? Please describe.

I’d like to enhance the existing documentation. ~~The current `README.md` houses one part of dev docs, that is going to be moved into the `/docs/developer` folder with #653~~ ✅ 

It might help contributors and extenders as well to have some additional docs about:

> [!NOTE]
> **Audit pass — 0.36.0.** Every box below was checked against the actual `docs/` tree rather than from memory, and each completed item now links to the file that covers it. Items discovered during the audit that were never on the list have been added, so this reflects what exists today. Gaps found along the way are collected under [Audit findings](#audit-findings). Most of the open boxes are closed by PR #2332.

#### Understanding GatherPress 

- [x] https://github.com/GatherPress/gatherpress-alpha — [developer/gatherpress-alpha](https://github.com/GatherPress/gatherpress/blob/develop/docs/developer/gatherpress-alpha/README.md) _(what it is, why migrations live in a companion plugin, the version lockstep, and both ways to apply updates)_
- [x] _close-to-core_ philosophy & community promise — [developer/philosophy](https://github.com/GatherPress/gatherpress/blob/develop/docs/developer/philosophy/README.md)
- [x] [`wpcli` commands](https://github.com/wp-cli/handbook) that GatherPress provides — [developer/wp-cli](https://github.com/GatherPress/gatherpress/blob/develop/docs/developer/wp-cli/README.md) _(added in the audit: every command, its options, and how to add one)_
- [x] Filters & Actions available in the plugin — [auto-generated hook reference](https://github.com/GatherPress/gatherpress/tree/develop/docs/developer/hooks) + [naming convention](https://github.com/GatherPress/gatherpress/blob/develop/docs/developer/hooks-naming-convention.md)
   - [x] Autoloader — covered by the generated reference
   - [x] Settings API — [settings/architecture.md](https://github.com/GatherPress/gatherpress/blob/develop/docs/developer/settings/architecture.md)
   - [x] Import & export — [settings/architecture.md#importexport](https://github.com/GatherPress/gatherpress/blob/develop/docs/developer/settings/architecture.md)
- [x] Implementation-Details on registered …
   - [x] post types — [post-type-supports/README.md](https://github.com/GatherPress/gatherpress/blob/develop/docs/developer/post-type-supports/README.md)
   - [x] post meta data — [post-type-supports/README.md](https://github.com/GatherPress/gatherpress/blob/develop/docs/developer/post-type-supports/README.md)
   - [x] taxonomies — [developer/data-model](https://github.com/GatherPress/gatherpress/blob/develop/docs/developer/data-model/README.md) _(added in the audit: all five, public and internal)_
   - [x] user meta data — [developer/data-model](https://github.com/GatherPress/gatherpress/blob/develop/docs/developer/data-model/README.md)
   - [x] settings — [settings/architecture.md](https://github.com/GatherPress/gatherpress/blob/develop/docs/developer/settings/architecture.md)
   - [x] REST endpoints — [developer/data-model](https://github.com/GatherPress/gatherpress/blob/develop/docs/developer/data-model/README.md) _(all nine, verified against the live route registry)_
- [x] URLs & permalinks (#744) — [developer/urls-and-permalinks](https://github.com/GatherPress/gatherpress/blob/develop/docs/developer/urls-and-permalinks/README.md) _(every public URL, verified against `wp rewrite list` on a running site; localized slugs, the untranslated `ical` slug, and rewrite flushing)_
- [x] custom DB table — [developer/data-model](https://github.com/GatherPress/gatherpress/blob/develop/docs/developer/data-model/README.md) _(schema, and why it is a table rather than post meta)_
- [x] Multisite compatibility — [developer/multisite](https://github.com/GatherPress/gatherpress/blob/develop/docs/developer/multisite/README.md) _(activation across sites, sites created and deleted later, settings inheritance, the two uninstall postures, and the network-wide user meta gotcha)_

#### Extending GatherPress 
 
- [x] Demo-data and prepared WordPress playgrounds — [developer/demo-data](https://github.com/GatherPress/gatherpress/blob/develop/docs/developer/demo-data/README.md) _(the two blueprints, the import shim, why the demo WXR is versioned per release, and how to add your own events)_
- [x] Import & Export events, venues & topics — [developer/content-import-export](https://github.com/GatherPress/gatherpress/blob/develop/docs/developer/content-import-export/README.md) _(added in the audit: how dates survive WXR, and the `gatherpress_pseudo_post_metas` filter for your own data)_
- [x] https://github.com/GatherPress/gatherpress-awesome — [developer/companion-plugins](https://github.com/GatherPress/gatherpress/blob/develop/docs/developer/companion-plugins/README.md) _(what a companion plugin is, Awesome as the template, and where to extend)_
- [x] Blocks
   - [x] hookable patterns (#888) — [blocks/hookable-patterns](https://github.com/GatherPress/gatherpress/blob/develop/docs/developer/blocks/hookable-patterns/README.md)
   - [x] Add or Remove blocks from the default event & venue templates using the Block Hooks API (#626) — [blocks/hookable-patterns](https://github.com/GatherPress/gatherpress/blob/develop/docs/developer/blocks/hookable-patterns/README.md). _Correction: my earlier "partial" note was wrong. Adding, removing and reordering were already documented with working examples. What was missing was a reference for the two anchors (pattern name, anchor block, position, and what is hooked by default), which has now been added._
   - [x] Editor slots available (#888) — [blocks/slot-fills](https://github.com/GatherPress/gatherpress/blob/develop/docs/developer/blocks/slot-fills/README.md)
   - [x] Styling event- & venue-blocks with `theme.json` — [theme-customizations](https://github.com/GatherPress/gatherpress/blob/develop/docs/developer/theme-customizations/README.md) (CSS custom properties, global styles fallbacks)
- [x] post-processing RSVPs (#787) — [rsvp/README.md — Acting on an RSVP](https://github.com/GatherPress/gatherpress/blob/develop/docs/developer/rsvp/README.md)
- [x] Overwrite templates from within a theme (#831) — [theme-customizations](https://github.com/GatherPress/gatherpress/blob/develop/docs/developer/theme-customizations/README.md)
- [x] Add custom URL endpoints (#831) — [custom-url-endpoints](https://github.com/GatherPress/gatherpress/blob/develop/docs/developer/custom-url-endpoints/README.md)
- [x] Add custom settings to GatherPress‘ settings tabs — [settings/extending-settings.md](https://github.com/GatherPress/gatherpress/blob/develop/docs/developer/settings/extending-settings.md)

#### Added during the audit — already covered

Docs that exist but were never on this list:

- [x] [RSVP system for developers](https://github.com/GatherPress/gatherpress/blob/develop/docs/developer/rsvp/README.md) — comment storage, statuses, providers, flags, comment-query coexistence
- [x] [`post_type_supports` extensibility](https://github.com/GatherPress/gatherpress/blob/develop/docs/developer/post-type-supports/README.md) — putting GatherPress features on your own post types
- [x] [Plugin lifecycle](https://github.com/GatherPress/gatherpress/blob/develop/docs/developer/plugin-lifecycle.md) — load sequence and `gatherpress_loaded`
- [x] [Event duration filters](https://github.com/GatherPress/gatherpress/blob/develop/docs/developer/event-duration.md)
- [x] [Venue address format](https://github.com/GatherPress/gatherpress/blob/develop/docs/developer/venue-address-format.md)
- [x] [Pattern pickers](https://github.com/GatherPress/gatherpress/blob/develop/docs/developer/blocks/pattern-pickers.md)
- [x] [Sharing utility styles](https://github.com/GatherPress/gatherpress/blob/develop/docs/developer/blocks/sharing-utility-styles.md)

### `docs/contributor/`

- [x] Code of conduct — [CODE_OF_CONDUCT.md](https://github.com/GatherPress/gatherpress/blob/develop/CODE_OF_CONDUCT.md), now linked from the new [contributor index](https://github.com/GatherPress/gatherpress/blob/develop/docs/contributor/README.md)
- [x] [Collaborator Access](https://github.com/GatherPress/gatherpress/blob/develop/docs/contributing.md) — "Access & Roles" in contributing.md
- [x] code contribution
   - [x] `wp-env` development environment — [developer/README.md](https://github.com/GatherPress/gatherpress/blob/develop/docs/developer/README.md)
   - [x] tools & workflows for quality standards
      - [x] screenshots for w.org (#822) — [screenshot-generator](https://github.com/GatherPress/gatherpress/blob/develop/docs/contributor/screenshot-generator/README.md)
      - [x] Version bump script by @mauteri — [release-process.md](https://github.com/GatherPress/gatherpress/blob/develop/docs/contributor/release-process.md)
      - [x] contributor- & credits script by @mauteri — [contributing.md — Contributor Credits](https://github.com/GatherPress/gatherpress/blob/develop/docs/contributing.md)
      - [x] e2e tests (#810) — [contributor/e2e-tests](https://github.com/GatherPress/gatherpress/blob/develop/docs/contributor/e2e-tests/README.md) _(added in the audit; the `test/e2e/README.md` it links to also described a page-object architecture that does not exist, and has been corrected)_
      - [x] unit tests (#831) — [contributor/unit-tests](https://github.com/GatherPress/gatherpress/blob/develop/docs/contributor/unit-tests/README.md)
      - [x] Prepare & publish a release — [release-process.md](https://github.com/GatherPress/gatherpress/blob/develop/docs/contributor/release-process.md)
   - [x] forking & git branch management (#948) — [contributor/README.md](https://github.com/GatherPress/gatherpress/blob/develop/docs/contributor/README.md) _(branches, squash-merge, changelog handling, in one place)_
- [x] no-code contribution
   - [x] Test actual PRs (#805) — [contributor/README.md](https://github.com/GatherPress/gatherpress/blob/develop/docs/contributor/README.md#test-a-pull-request)
   - [x] Add your events to the demo-data — [contributor/README.md](https://github.com/GatherPress/gatherpress/blob/develop/docs/contributor/README.md#add-to-the-demo-data) (the repo is [gatherpress-demo-data](https://github.com/GatherPress/gatherpress-demo-data))
   - [x] Translate the plugin — [contributor/README.md](https://github.com/GatherPress/gatherpress/blob/develop/docs/contributor/README.md#translate)
   - [x] Join the weekly huddle & the slack channel — [contributor/README.md](https://github.com/GatherPress/gatherpress/blob/develop/docs/contributor/README.md#show-up)
   - [x] Help Triaging tickets — [contributor/README.md](https://github.com/GatherPress/gatherpress/blob/develop/docs/contributor/README.md#triage-issues)
   - [x] Spread the word in social media & blog about GatherPress — [contributor/README.md](https://github.com/GatherPress/gatherpress/blob/develop/docs/contributor/README.md#talk-about-it)
   - [x] ~~Take part in the pilot program and run your meetup with demo.gatherpress.org~~ _(dropped: no longer relevant)_
- [x] An index for `docs/contributor/` — [contributor/README.md](https://github.com/GatherPress/gatherpress/blob/develop/docs/contributor/README.md) _(added in the audit; it doubles as the Contribution Guidelines entry point)_

### `docs/user/`

- [x] Install — [installation.md](https://github.com/GatherPress/gatherpress/blob/develop/docs/installation.md) / [getting-started.md](https://github.com/GatherPress/gatherpress/blob/develop/docs/user/getting-started.md)
- [x] Settings — [configuration.md](https://github.com/GatherPress/gatherpress/blob/develop/docs/configuration.md)
- [x] Create a new event — [creating-and-managing-events.md](https://github.com/GatherPress/gatherpress/blob/develop/docs/user/creating-and-managing-events.md)
- [x] Create a new venue — [venues.md](https://github.com/GatherPress/gatherpress/blob/develop/docs/user/venues.md)
- [x] Event topics — [topics.md](https://github.com/GatherPress/gatherpress/blob/develop/docs/user/topics.md)
- [x] Send Emails — [user/emails.md](https://github.com/GatherPress/gatherpress/blob/develop/docs/user/emails.md)
- [x] Export to and subscribe with your calendar app (#831) — [user/calendar-feeds.md](https://github.com/GatherPress/gatherpress/blob/develop/docs/user/calendar-feeds.md)
- [x] Blocks overview — [user/blocks/README.md](https://github.com/GatherPress/gatherpress/blob/develop/docs/user/blocks/README.md), now linking every documented block
   - [x] Event Query block — [user/blocks/event-query.md](https://github.com/GatherPress/gatherpress/blob/develop/docs/user/blocks/event-query.md)
   - [x] Event Date block — [user/blocks/event-date.md](https://github.com/GatherPress/gatherpress/blob/develop/docs/user/blocks/event-date.md)
   - [x] RSVP block and inner blocks — [user/blocks/rsvp-and-inner-blocks.md](https://github.com/GatherPress/gatherpress/blob/develop/docs/user/blocks/rsvp-and-inner-blocks.md)
   - [x] RSVP Response block and inner blocks — [user/blocks/rsvp-response-and-inner-blocks.md](https://github.com/GatherPress/gatherpress/blob/develop/docs/user/blocks/rsvp-response-and-inner-blocks.md)
   - [x] Venue block (v2) — [user/blocks/venue-v2/README.md](https://github.com/GatherPress/gatherpress/blob/develop/docs/user/blocks/venue-v2/README.md)
   - [x] Add to Calendar block — [user/blocks/add-to-calendar.md](https://github.com/GatherPress/gatherpress/blob/develop/docs/user/blocks/add-to-calendar.md)
   - [x] Online Event block — [user/blocks/online-event.md](https://github.com/GatherPress/gatherpress/blob/develop/docs/user/blocks/online-event.md)

#### Added during the audit — already covered

- [x] [Core concepts](https://github.com/GatherPress/gatherpress/blob/develop/docs/user/core-concepts.md)
- [x] [RSVP system](https://github.com/GatherPress/gatherpress/blob/develop/docs/user/rsvp-system.md)
- [x] [User roles and permissions](https://github.com/GatherPress/gatherpress/blob/develop/docs/user/user-roles-and-permissions.md)
- [x] [Privacy](https://github.com/GatherPress/gatherpress/blob/develop/docs/user/privacy.md)
- [x] [Event ID override](https://github.com/GatherPress/gatherpress/blob/develop/docs/user/event-id-override.md)
- [x] [Where to get help](https://github.com/GatherPress/gatherpress/blob/develop/docs/user/where-to-get-help.md)
- [x] [FAQ](https://github.com/GatherPress/gatherpress/blob/develop/docs/faq.md)

### Audit findings

Problems the audit turned up that are not "a doc is missing":

- [x] **Four broken internal links** in `docs/` — all fixed:
   - `docs/playground.md` → `docs/contributor/playground-pr-preview/README.md` (×2) — path was written from the repo root but resolves from inside `docs/`
   - `docs/user/core-concepts.md` → `./blocks.md` — the file is `./blocks/README.md`
   - `docs/contributor/screenshot-generator/README.md` → `../e2e-tests` — the target has now been written
- [x] **`wp gatherpress event rsvp` silently rejected two of its own options.** The command body read `--guests` and `--anonymous`, but neither appeared in the `## OPTIONS` synopsis, so WP-CLI refused them (`unknown --guests parameter`) and that code was unreachable. Both are now declared, verified end to end: they persist, clamped by the event's own `gatherpress_max_guest_limit` and `gatherpress_enable_anonymous_rsvp` exactly as the front-end form is.
- [x] **No `docs/README.md`.** Added — a top-level index tying together `user/`, `developer/` and `contributor/`.

### Designs

For now, I recommend, creating static markdown files until a new Playground-powered documentation workflow #765 is ready. A structure with named folders and just one README.md file per directory seems to be the most common way to do that.

The written docs should be accompanied by an auto-generated code reference. 

Staying with GatherPress‘ _close-to-core_ philosophy, I’d like to re-use existing stuff and get the benefit of a well-known UI on top. A working PoC for a GatherPress code reference using the same theme as `developer.wordpress.org`, can be found here https://github.com/carstingaxion/gatherpress-devhub/issues/1#issuecomment-2294166369

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by carstingaxion** [comment]
https://mermaid.live/edit#pako:eNplj7GuwjAMRX_F8vAmyvDGDkjwgBkBG2GwGtNWNEnlJC2o9N8JTx2Q6sm-50jWHbBwmjHHW-P6oiIJcN4qC2nWQ_QsoF0xQpatXg_TvGBim0Fz90F-zv4SMw40BRqn5Mu5HBp6luKi1bB0Ul5nyu5bORyhFe5q7ufi_tK3GdsOfoB_GQL74P8tXKBhMVTrVGz4JApDxYYV5mnVJHeFyo7Joxjc6WkLzINEXmD6WlaY36jx6YptasHbmkohM6XjG8dxZLI

**Comment by carstingaxion** [comment]
May I invite you too @oscarhugopaz, as you seem to do more development work, than a typical user, would you please add your thoughts here?!

What needs to be part of GatherPress developer documentation?

**Comment by shawfactor** [comment]
I’ve never looked at the documentation but I managed to understand the code. My only only bugbear was that many methods are protected which are useful for building templates and other plugin extensions. I’ve raised an issue for this already though as there are no security reasons why these methods are not accessible to everyone

**Comment by carstingaxion** [comment]
> I’ve never looked at the documentation but I managed to understand the code.

Thanks @shawfactor, in the name of all who wrote that understandable code!

I felt & feel the same, that’s why I really enjoy contributing to that project! The existing code base is understandable and makes a lot of sense. Maybe we could keep your statement for a later, as a first testimonial. Accompanying your great review on w.org ;)

> My only only bugbear was that many methods are protected which are useful for building templates and other plugin extensions. I’ve raised an issue for this already though as there are no security reasons why these methods are not accessible to everyone

We talked about your issue #635 in last weeks huddle and concluded that the current implementation is based more on a general design decision, rather than being planned for extension. But we agree and are willing to change some of the methods. 

Be absolutely invited to open PRs for all changes you see a need for!

**Comment by shawfactor** [comment]
Completely agree with you.

I swapped my sites over from another events calendar as gatgerpress does
things in a true open source way, seems open to suggestions, and plans to
work with existing Wordpress plugins

Happy for you to use my words as a testimonial

I believe having accessible methods or some alternatives like template
functions is essential though. Buddypress, bbpress, woocommerce etc don’t
try to do everything. Instead they benefit from an ecosystem that has grown
with them. In order to achieve this, accessing event characteristics like
times and descriptions in a flexible way (without having to reinvent the
wheel) is essential.

Pere

**Comment by carstingaxion** [comment]
The index of `docs/contributor` could become also the _Contribution Guidelines_ GitHub wants us to add.

![image](https://github.com/user-attachments/assets/9808f884-8b43-42fd-8926-20042b2c7ddf)

**Comment by MervinHernandez** [comment]
QUEUE for Discussion - October 25
- Mervin - strongly suggests breaking up this documentation ticket into smaller tickets for delegating / better tracking.
-- Split 1 - section "x" of documentation
-- Split 2 - connecting our dev docs to git required "Contributor Guidelines"
- ?? Should this be split into a ticket for (more) changing of protected methods?

**Comment by MervinHernandez** [comment]
Queued to break this apart.

**Comment by MervinHernandez** [comment]
Queued to break this apart.

**Comment by mauteri** [comment]
Adding this to 0.36.0 as a documentation audit. @carstingaxion put this ticket in over 2 years ago and we've made a lot of strides improving documentation. @patriciabt @supernovia maybe we can all review this and make sure there are no gaps. Thx!

**Comment by mauteri** [comment]
Took a big swing at this one and closed out the remaining boxes.

I worked with Claude on it and read over the output. It was a large PR, so if you spot something missing or inaccurate, please don't hesitate to open a follow up. That's very welcome.

Everything landed in #2332.

## How the audit ran

Every box above was checked against the actual `docs/` tree rather than from memory, and each completed item now links to the file that covers it, so the check marks are verifiable rather than a claim. A number of things were already documented but had never been added to this list, so those got added and checked. The pilot program line is dropped since it's no longer relevant.

## What got written

Developer docs:

- **Data model**: post types, taxonomies, post/comment/user meta, the `gatherpress_events` table, and every REST route
- **Multisite**: what's per site and what's network wide, from activation through uninstall
- **URLs and permalinks**: every public URL, where its slug comes from, and when rewrite rules flush
- **WP-CLI commands**: the full command surface and how to add one
- **Content import and export**: how event dates survive a WXR round trip via `gatherpress_pseudo_post_metas`
- **Close to core**: the design philosophy, and where the line falls between core and a companion plugin
- **Companion plugins**: building on GatherPress, starting from GatherPress Awesome
- **GatherPress Alpha**: what it is and why it's version locked to core
- **Demo data and prepared playgrounds**: the blueprints and where they're used

Contributor docs:

- **End to end tests**: the guide `screenshot-generator` had been linking to since #822. It was never written
- **An index for `docs/contributor/`**, which doubles as the Contribution Guidelines entry point @carstingaxion asked for in a comment above. Covers the branch model, the test suites, finding something to work on from the current milestone, and the no code contributions that had nowhere to live: reviewing a PR preview, triage, translation, demo data
- **A top level `docs/README.md`**, since there was no entry point tying user, developer and contributor together

User docs: Event Date, Add to Calendar and Online Event blocks.

## Things the audit turned up

Not everything was a missing page.

**Four broken internal links.** One of them was `screenshot-generator` pointing at `../e2e-tests`, which had never existed.

**`test/e2e/README.md` described an architecture that isn't there.** It documented a page object model with `pages/EventPage.js` and `fixtures/TestDataFactory.js`, and told contributors to require them. None of those files exist and every spec requires only `@playwright/test`, so anyone following it wrote a test that couldn't run. It now describes the real suite.

**`wp gatherpress event rsvp` was rejecting two of its own options.** The command reads `--guests` and `--anonymous`, but neither was declared in its `## OPTIONS` synopsis, so WP-CLI refused them outright and that code was unreachable. Both are declared now and verified end to end. That's the only behavior change in the PR and it has a changelog entry.

One correction worth recording: I first marked the Block Hooks item as only partly covered. That was wrong. Adding, removing and reordering were already documented with working examples. What was actually missing was a reference for the two anchors, so that table got added instead.

## Still worth doing

The code reference from the Designs section is still open, and #765 is still the tracking issue for a Playground powered docs workflow. Neither is blocked by anything here.
